rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -74,11 +86,12 @@ Node::create_publisher(
   if (!allocator) {
     allocator = std::make_shared<Alloc>();
   }
+
   return rclcpp::create_publisher<MessageT, Alloc, PublisherT>(
     this->node_topics_.get(),
-    topic_name,
+    extend_name_with_sub_namespace(topic_name, this->get_sub_namespace()),
```

Comment:
Could we just get rid of `get_sub_namespace()` and get the private variable `this->sub_namespace` directly?  I think that would help with my concerns about the API being confusing (though it would make the tests somewhat less detailed).  Thoughts?

Reply:
I would not, as I said above, I don't see a reason not to allow access to it, and making the tests harder to implement seems like a pretty significant cost to me.